kernel/src/log_compaction/writer.rs

```diff
@@ -0,0 +1,177 @@
+use std::sync::Arc;
+
+use crate::action_reconciliation::RetentionCalculator;
+use crate::log_segment::LogSegment;
+use crate::path::ParsedLogPath;
+use crate::snapshot::Snapshot;
+use crate::table_properties::TableProperties;
+use crate::{DeltaResult, Engine, Error, Version};
+
+use url::Url;
+
+use super::COMPACTION_ACTIONS_SCHEMA;
+use crate::checkpoint::log_replay::{CheckpointBatch, CheckpointLogReplayProcessor};
+use crate::engine_data::FilteredEngineData;
+use crate::log_replay::LogReplayProcessor;
+
+/// Determine if log compaction should be performed based on the commit version and
+/// compaction interval.
+pub fn should_compact(commit_version: Version, compaction_interval: Version) -> bool {
```

Comment:
should we directly make this `pub` or instead expose this under a utility on `TableProperties` or similar? Would be happy to just make `pub(crate)` for now and take a fast-follow to expose some utility?

Reply:
This is part of the public API we want to expose to engines.  There is an equivalent function in the Java kernel.